### ISSUE TITLE
Create Big Cannons pre-assembled cannon cannot be mounted on Eureka ship

### ISSUE THREAD

**Issue by Daniel366Cobra** [ValkyrienSkies/Valkyrien-Skies-2#972]
### Mod Name

VS2 + Eureka + Create Big Cannons

### This issue occurs when only Valkyrien Skies, addons, and the mod I have specified are installed and no other mods

- [X] I have tested this issue and it occurs with only Valkyrien Skies, addons, and the mod I have specified

### Minecraft Version

1.20.x

### Mod Loader

Forge

### Issue description

- Attempting to assemble the ship with mounted pre-assembled cannons via Eureka helm leads to cannons detaching from their mounts and remaining in block form.
- Attempting to move the cannon onto the mount of assembled ship and attach it by activating the mount does not work.
- Only hand-building the cannon on the mount of already assembled ship works, but in survival building cannons from parts needs silk touch and welding and only works with bronze or cast iron cannons.

Mod versions:
Valkyrien Skies 2.3.0-beta.5
Eureka 1.5.1-beta.3
Create Eureka 1.0.0
Create Big Cannons 5.5.1

### Issue reproduction

- Build a structure equipped with a C: BC cannon on a cannon mount and a Eureka helm
- Assemble the structure into a ship via helm
- Observe cannon being left behind in block form

Additionally try powering the mount to assemble the cannon onto the mount before assembling the ship, the cannon will turn into block form and detach.

To test the "working" solution: assemble ship, then build the cannon piece-by-piece onto the mount. Assemble the cannon via mount.

**Comment by thetheaplant** [comment]
Create contraptions are entities, and thus not assembled with ships. Dissemble create contraptions before assembling them onto ships.

**Comment by Daniel366Cobra** [comment]
> Create contraptions are entities, and thus not assembled with ships. Dissemble create contraptions before assembling them onto ships.

I did not initially try to assemble the assembled contraption onto the ship. I had everything in block form - cannon on mount and the ship. After assembling the ship the cannon is left in the air in block form, not attached to the mount which gets assembled onto the ship.

I tried another way to keep the cannon on - placing a temporary block between the cannon and mount, right under the place where the cannon gets its horizontal axis, so that there is no air block between the ship and cannon. It worked in the sense that the cannon got attached to the ship, but the mount was unable to assemble the cannon saying "The cannon must have one and only one opening and closed end".
